### PR TITLE
adding swag button

### LINKED ISSUE #48208: Swag Button for RC
**Context**
Shiny new button in the admin app. For VIPs only.

[Slack Thread](https://metaboat.slack.com/archives/C07NW0WQG30/p1727358748619939)

### PR DESCRIPTION
**Pull request by npfitz**
Closes https://github.com/metabase/metabase/issues/48208

### Description
Adding a button in the admin section that should only show if the following is true:
- The version is a release candidate
- The date is before November 1st

Clicking it should open a modal that helps navigate you to the swag form.

### How to verify

Describe the steps to verify that the changes are working as expected.
1. Edit your `version.properties` folder to have version v0.51.3-rc
2. Start the BE, and navigate to the admin app
3. Witness the shiny button. Click it. You know you want to
4. A modal should appear with a link at the bottom. Clicking it should take you to a form, and navigating back to the app the modal should be gone and the button should be less shiny.

### Demo
chrome_4sdhB7prng

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/admin/settings/components/Swag/SwagButton.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  const version = useSetting("version");
+  const [modalOpen, setModalOpen] = useState(false);
+
+  // Need to check the value again when modal closes to remove animation
+  // So the modal opening / closing is a dependency here
+  const isLinkUsed = useMemo(() => {
+    const rawValue = localStorage.getItem(SWAG_51_LOCAL_STORAGE_KEY);
+    return rawValue === "true";
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [modalOpen]);
```
Comment (iethree) [inline frontend/src/metabase/admin/settings/components/Swag/SwagButton.tsx:23]: huh. funny that this complains about extra deps

File: frontend/src/metabase/admin/settings/components/Swag/utils.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export const isSwagEnabled = (version?: string) => {
+  if (!version) {
+    return false;
+  }
+
+  const CUTOFF_DATE = new Date("2024-10-01");
```
Comment (iethree) [inline frontend/src/metabase/admin/settings/components/Swag/utils.ts:6]: we need to change this before merging. Maybe Nov 1? @luizarakaki ?

File: frontend/src/metabase/admin/settings/components/Swag/SwagModal.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+import { getCurrentUser } from "metabase/admin/datamodel/selectors";
+import { useSelector } from "metabase/lib/redux";
+import { Button, Flex, Modal, type ModalProps, Text } from "metabase/ui";
+
+import { SWAG_51_LOCAL_STORAGE_KEY, SWAG_LINK } from "./constants";
+
+export const SwagModal = (props: Pick<ModalProps, "opened" | "onClose">) => {
+  const user = useSelector(getCurrentUser);
+  const url = user.email ? `${SWAG_LINK}?email=${user.email}` : SWAG_LINK;
```
Comment (iethree) [inline frontend/src/metabase/admin/settings/components/Swag/SwagModal.tsx:12]: did we decide against sending the instance uuid?

File: frontend/src/metabase/css/admin.module.css [hunk 4]
```diff
@@ -169,3 +169,38 @@
...
+  background-size: 400% 400%;
+  animation: animation-name 6s ease infinite;
+  border-radius: 0.5rem;
+  margin-left: 1rem;
+  margin-right: 1rem;
+  color: var(--mb-color-text-white);
+  cursor: pointer;
+}
+
+.LameSwag {
```
Comment (iethree) [inline frontend/src/metabase/css/admin.module.css:203]: 🤣  ❤️

File: frontend/src/metabase/admin/settings/components/Swag/SwagButton.unit.spec.tsx
Comment (iethree) [inline frontend/src/metabase/admin/settings/components/Swag/SwagButton.unit.spec.tsx:1]: good call not e2e testing this one. that would be a waste. 👍

File: frontend/src/metabase/admin/settings/components/Swag/SwagButton.unit.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  jest.useFakeTimers({
+    now: new Date("2024-09-27"),
+  });
+  setup();
+  expect(screen.getByText(/Claim your swag/i)).toBeInTheDocument();
+});
+
+it("should not render a button when date is above threshold", () => {
+  jest.useFakeTimers({
+    now: new Date("2024-10-27"),
```
Comment (iethree) [inline frontend/src/metabase/admin/settings/components/Swag/SwagButton.unit.spec.tsx:39]: maybe paramaterize this? cuz you're gonna have to change all these tests

**Review by iethree — APPROVED** [review]
Looks great! 👕

File: frontend/src/metabase/admin/settings/components/Swag/SwagModal.tsx (hunk 3 above)
Comment (npfitz) [reply frontend/src/metabase/admin/settings/components/Swag/SwagModal.tsx:12]: It was mentioned, but I don't think we are going to bother.

https://metaboat.slack.com/archives/C07NW0WQG30/p1727404321295939?thread_ts=1727402929.582489&cid=C07NW0WQG30

**Pull request merged**